processor/src/host/mod.rs

```diff
@@ -82,6 +82,15 @@ pub trait Host {
         Ok(HostResponse::None)
     }
 
+    /// Handles the failure of the assertion instruction.
+    fn on_assert_failed<S: ProcessState>(&mut self, process: &S, err_code: u32) -> ExecutionError {
+        ExecutionError::FailedAssertion {
+            clk: process.clk(),
+            err_code,
+            err_msg: "Unknown error",
+        }
```

Comment:
Should we change the error message to `Option<String>` and here just pass `None`?

Reply:
Right, that would be better.